File: primes.rb
def prime(numInt)
	isPrime = true
	for x in 2..numInt-1
		if numInt % x == 0 then
			isPrime = false
			break
		end
	end
	return isPrime	
end

def printPrimes(primes)
	puts "Primes are: "
	primes.each do |i|
		puts i
	end
end

primes = Array.new
while true do
	
	puts "Enter integer"
	numString = gets
	numInt = numString.to_i
	if numInt == 0 then
		break
	end
	isPrime = prime(numInt)
	if isPrime == true then
		primes.push(numInt)
	end
end

printPrimes(primes)
	

File: primes.py
def prime(numInt):
	isPrime = True
	if numInt != 1:
		for x in range(2,numInt):
			if numInt % x == 0:
				isPrime = False
				break
	else:
		isPrime = False
	return isPrime	
	
def print_primes(primes):
		print "\nPrimes are: "
		for p in primes:
			print p
 

def main():
	primes = []
	while True:
		print "Enter a number: "
		numString = raw_input()
		numInt = long(numString)
		if numInt == 0:
			break
		isPrime = prime(numInt)
		if isPrime == True:
			primes.append(numInt)
			
	print_primes(primes)
	
	

if __name__ == '__main__':
	main()

